### PR TITLE
spaces-defaults

### LINKED ISSUE #674: Cache examples by default on Spaces
Examples should be cached by default on spaces. Much better user experience. 

Ideally the way to do it would be by reading from an env variable that spaces can provide. 

We could also potentially cache them by default everywhere, but to avoid  slowing down the interface launch we'd want to cache as the first users click them instead of during the launch event.

### PR DESCRIPTION
**Pull request by FarukOzderim**
- enable_queue and cache_examples are activated in spaces in default option

fixes #674 #680

### CODE REVIEW

File: gradio/blocks.py [hunk 1]
```diff
@@ -433,7 +433,10 @@ def launch(
                 getpass.getpass("Enter key for encryption: ")
             )
 
-        self.enable_queue = enable_queue
+        if self.is_space and enable_queue is None:
+            self.enable_queue = True
+        else:
+            self.enable_queue = enable_queue
```
Comment (abidlabs) [inline gradio/blocks.py:439]: What happens if `enable_queue` is not provided and you are *not* running in Spaces? Should explicitly make the default `False`

```suggestion
            self.enable_queue = enable_queue or False
```

File: gradio/interface.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ def __init__(
         outputs: str | Component | List[str | Component] = None,
         verbose: bool = False,
         examples: Optional[List[Any] | List[List[Any]] | str] = None,
-        cache_examples: bool = False,
+        cache_examples: Optional[bool] = None,
```
Comment (abidlabs) [inline gradio/interface.py:112]: Same as above, let's update the docstring

File: gradio/interface.py [hunk 3]
```diff
@@ -288,7 +288,6 @@ def clean_html(raw_html):
 
         self.thumbnail = thumbnail
         theme = theme if theme is not None else os.getenv("GRADIO_THEME", "default")
-        self.is_space = True if os.getenv("SYSTEM") == "spaces" else False
```
Comment (abidlabs) [inline gradio/interface.py:291]: Good catch of duplication

File: gradio/interface.py [hunk 4]
```diff
@@ -487,8 +486,11 @@ def clean_html(raw_html):
             if component.label is None:
                 component.label = "output_" + str(i)
 
-        self.cache_examples = cache_examples
-        if cache_examples:
+        if self.is_space and cache_examples is None:
+            self.cache_examples = True
+        else:
+            self.cache_examples = cache_examples
```
Comment (abidlabs) [inline gradio/interface.py:492]: Same as above, consider

```suggestion
            self.cache_examples = cache_examples or False
```

**Comment by abidlabs** [comment]
Looks good! Minor points above

File: gradio/interface.py (hunk 2 above)
Comment (FarukOzderim) [reply gradio/interface.py:112]: Wow it cache_examples was missing in the docstring :D

**Comment by FarukOzderim** [comment]
@abidlabs thanks for the careful review!

**Pull request merged**